pkg/dockerregistry/server/quotarestrictedblobstore.go

```diff
@@ -122,7 +123,51 @@ func (bw *quotaRestrictedBlobWriter) Commit(ctx context.Context, provisional dis
 
 // admitBlobWrite checks whether the blob does not exceed image quota, if set. Returns
 // ErrAccessDenied error if the quota is exceeded.
-func admitBlobWrite(ctx context.Context, repo *repository) error {
+func admitBlobWrite(ctx context.Context, repo *repository, size int64) error {
+	if size < 1 {
+		return nil
+	}
+
+	err := admitLimitRanges(ctx, repo, size)
+	if err != nil {
+		return err
+	}
+
+	err = admitQuotas(ctx, repo)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// admitLimitRanges checks the blob against any established limit ranges.
+func admitLimitRanges(ctx context.Context, repo *repository, size int64) error {
+	limitranges, err := repo.limitClient.LimitRanges(repo.namespace).List(kapi.ListOptions{})
+	if err != nil {
+		if kerrors.IsForbidden(err) {
+			context.GetLogger(ctx).Warnf("Cannot list limitranges because of outdated cluster roles: %v", err)
```

Comment:
Reconciling roles is (or if it's not - should be) a *required* step during upgrade. Not executing it will give you unpredictable errors and we should warn admins about it in every upgrade doc. @deads2k @derekwaynecarr and @liggitt ensured me about my worries regarding swallowing this error. I don't see the need for yet another tool/check other than this, which if you as an admin forgot to run reconcile you will forget about the other as well.